### DISCUSSION TITLE
Can I provide built binaries for other MacOS 11 users?

### DISCUSSION THREAD

**Discussion by SolsticeSpectrum** [Homebrew/discussions#5582] [Casks]
### Description of issue

I have no idea if I picked correct category but I had some struggle with compiling openssl@3 on MacOS 11 running on old 2013 macbook. After several tries, it worked but it took looooong time because it compiles very slowly. So is there a way I can help others by providing the compiled binaries so others don't have to compile this version?

### Relevant casks

openssl

**Comment by ZhongRuoyu** [comment]
You can run `brew install --build-bottle openssl@3`, and then `brew bottle openssl@3`. You will then have a bottle, with which you can do `brew install path/to/bottle.tar.gz` on compatible machines.

Since you said you want to share the binaries, do note that it's your responsibility to make sure and convince others that the bottle is functional, safe and not tampered with. Especially taking into account that `openssl@3` is relied on by a lot of formulae and (very) bad things can happen if it's broken.

**Reply by trinitronx** [reply]
> You will then have a bottle, with which you can do `brew install path/to/bottle.tar.gz` on compatible machines.

It looks like this no longer works without `HOMEBREW_DEVELOPER=1`?

```console
vagrant@default-macos-12-1-lyraphase-runner bottles-artifact % brew install ./xz--5.8.1.monterey.bottle.tar.gz 
Warning: No available formula with the name "xz--5.8.1.monterey.bottle.tar.gz".
==> Searching for similarly named formulae and casks...
Error: No formulae or casks found for xz--5.8.1.monterey.bottle.tar.gz.

vagrant@default-macos-12-1-lyraphase-runner bottles-artifact % brew install --formula ./xz--5.8.1.monterey.bottle.tar.gz
Warning: No available formula with the name "xz--5.8.1.monterey.bottle.tar.gz".
==> Searching for similarly named formulae...
Error: No formulae found for xz--5.8.1.monterey.bottle.tar.gz.


vagrant@default-macos-12-1-lyraphase-runner bottles-artifact % export HOMEBREW_DEVELOPER=1; brew install --formula ./xz--5.8.1.monterey.bottle.tar.gz
==> Fetching downloads for: xz
==> Installing xz from lyraphase/right2repair
==> Pouring xz--5.8.1.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/xz/5.8.1: 96 files, 2.5MB
==> Running `brew cleanup xz`...
Disable this behaviour by setting `HOMEBREW_NO_INSTALL_CLEANUP=1`.
Hide these hints with `HOMEBREW_NO_ENV_HINTS=1` (see `man brew`).
==> No outdated dependents to upgrade!
```

**Reply by p-linnane** [reply]
Correct. Formulae are required to be in a tap now.
